userguide/one_shot_object_detection/README.md

```diff
@@ -0,0 +1,120 @@
+
+augmented_images.explore()
+```
+
+![Explore augmented images](images/augmented_images_explore.jpg)
+
+Some examples of synthetic training data generated:
+
+![Sample augmented images](images/augmented_images_collage.jpg)
+
+*Note: In order to view the exact augmented images used to train the model, the same seed should be used which was used to train the model.*
+
+
+To visualize the predictions made on the test set:
+
+
+```python
+test_data['image_pred'] = tc.object_detector.util.draw_bounding_boxes(test_data['image'], test_data['predictions'])
+
+test_data.explore()
+```
+
+![Prediction test data](images/test_data_explore.jpg)
+
+Examples of test image predictions:
+
+![Sample prediction image - 1 and 2](images/sample_prediction_images_1_2.jpg)
+
+
+#### How it works
+
+To begin training for one-shot object detector, a set of synthetic training data needs to be generated using the provided starter training image(s). These images are augmented with operations: horizontal flip, vertical flip, yaw, pitch, roll, blur, noise and lightening perturbations. These operations are applied on a set of background images. The background images may be provided by the user. If no background images are provided, a set of default background images are downloaded and used.
+
+To learn more about the training implementation details, refer to the Object Detector: How it Works (https://apple.github.io/turicreate/docs/userguide/object_detection/how-it-works.html) chapter.
+
+
+#### Deployment to CoreML
+
+To learn more about deploying One-Shot to CoreML details, refer to the Object Detector: Deployment to Core ML (https://apple.github.io/turicreate/docs/userguide/object_detection/export-coreml.html) chapter.
+
+
+#### Advanced Usage
+
+**Background images**
+
+The One-Shot Object Detector generates a set of synthetic images with which to train the detector. These images contain the object in various transformations in the wild. When the network is trained with these images, it allows the model to learn to identify the object in different environments. For convenience, we provide a set of background images by default. However, you may also supply your own custom background images specific to your application. These can be supplied within the `create()` method as follows:    
+
+```python
+import turicreate as tc
+
+# Load the starter images
+starter_images = tc.SFrame({'image':[tc.Image('stop_sign.png')],
+                   'label':['stop_sign']})
+
+# Load background images
+my_backgrounds = tc.SArray('my_custom_backgrounds.sarray')
+
+# Create a model using custom background images                                    
+model = tc.one_shot_object_detector.create(starter_images, 'label', backgrounds = my_backgrounds)
+```
+
+To view the synthetic images generated with your custom background images:
+
+```python
+augmented_images = tc.one_shot_object_detector.util.preview_synthetic_training_data(starter_images, 'label', my_backgrounds)
```

Comment:
synthetic_images?

Reply:
This is to remain consistent with your naming convention in `create()`:

`augmented_data = _preview_synthetic_training_data(data, target, backgrounds)`

Let's agree on the naming convention we want to use in the toolkit so I can reflect that in the user guide.

Reply:
Let's use the word `synthetic` everywhere instead of `augmented`.